app/code/Magento/Inventory/Model/Reservation/ReservationBuilder.php

```diff
@@ -109,6 +109,7 @@ public function setMetadata(string $metadata = null): ReservationBuilderInterfac
     public function build(): ReservationInterface
     {
         $data = [
+            ReservationInterface::RESERVATION_ID => null,
```

Comment:
The reason is that If we don't pass the reservation ID, the constructor (used by the object manager) will throw an exception because it's a [constructor mandatory parameter](https://github.com/magento-engcom/msi/blob/develop/app/code/Magento/Inventory/Model/Reservation.php#L42)